+sw_tests/+utilities/mock_function.m

```diff
@@ -14,7 +14,7 @@
                      'function varargout = %s(varargin)\n' ...
                      '    persistent n_calls;\n' ...
                      '    persistent arguments;\n' ...
-                     '    if ischar(varargin{1}) && strcmp(varargin{1}, ''check_calls'')\n' ...
+                     '    if nargin > 0 && ischar(varargin{1}) && strcmp(varargin{1}, ''check_calls'')\n' ...
```

Comment:
Thanks for fixing this... it's really annoying that Matlab is so _functions_-based but they only provide a mock framework for _classes_. This implementation is a bit fragile though and fails if the user doesn't have write access to the current working directory (which could have been changed by the code being tested!)... for a future maintenance task it might be worthwhile to refactor it to create the function file in a temporary folder and then to add this folder to the top of the path with an `onCleanup` to remove the folder from the path and delete it when everything is done.

Reply:
Done https://github.com/SpinW/spinw/issues/94